STRETCH.md

```diff
@@ -3,21 +3,136 @@
 Stretch challenges are for those who are already familiar with web3 development or want a more significant challenge. You may also start with the stretch challenges immediately if you are working in teams. In that case, try the challenges of your respective track before the workshop.
 
 - [Stretch Challenges](#stretch-challenges)
-  - [Stretch Challenge 1: Reantrancy Attacks 👾](#stretch-challenge-1-reantrancy-attacks-)
-
-## Stretch Challenge 1: Reantrancy Attacks 👾
-
-- Reentrancy attacks
-- Testing frontend
-- Frontend mint without wagmi / rainbowkit
-- NFT pokemon clone
-- Randomness games such as dice game (Speedrun Ethereum level 3) or a lottery
-- Blockchain tic tac toe
-- Mint that costs money and has royalties
-- Staking
-- Gas optimization
-- ERC1155/ERC721A/ERC721Psi
-- Whitelisting with Merkle Trees
-- Meta transaction with ....
-- Rentable NFT with EIP-4907
-- NFT Gated access
+  - [Stretch Challenge 1: Reentrancy Attacks 👾](#stretch-challenge-1-reentrancy-attacks-)
+  - [Stretch Challenge 2: Testing the Frontend](#stretch-challenge-2-testing-the-frontend)
+  - [Stretch Challenge 3: Minting Dapp Without Wagmi/RainbowKit](#stretch-challenge-3-minting-dapp-without-wagmi--rainbowkit)
+  - [Stretch Challenge 4:](#stretch-challenge-4-nft-pokemon-clone)
+  - [Stretch Challenge 5: Dice Game](#stretch-challenge-5-dice-game)
+  - [Stretch Challenge 6: Blockchain Tic-Tac-Toe](#stretch-challenge-6-blockchain-tic-tac-toe)
+  - [Stretch Challenge 7: Payable Mint, Royalties, and Withdrawing](#stretch-challenge-7-payable-mint-royalties-and-withdrawing)
+  - [Stretch Challenge 8: Staking](#stretch-challenge-8-staking)
+  - [Stretch Challenge 9: Gas Optimization](#stretch-challenge-9-gas-optimization)
+  - [Stretch Challenge 10: Whitelisting/Allowlisting with Merkle Trees](#stretch-challenge-10-whitelistingallowlisting-with-merkle-trees)
+  - [Stretch Challenge 11: Meta Transactions](#stretch-challenge-11-meta-transactions)
+  - [Stretch Challenge 12: Rentable NFTs](#stretch-challenge-12-rentable-nfts)
+  - [Stretch Challenge 13: NFT Gated Access](#stretch-challenge-13-nft-gated-access)
```

Comment:
Did you know you can use a Markdown plugin to generate these automatically? Then, if you change a heading, the TOC will automatically be updated too. See:  https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one